### PR TITLE
Enable revive linter

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
**What this PR does / why we need it**:
Enables the revive linter:
> Fast, configurable, extensible, flexible, and beautiful linter for Go. Drop-in replacement of golint.
 - https://golangci-lint.run/usage/linters/#revive

Golint itself has been deprecated for 3 years.

**Special notes for your reviewer**:
PR #3193 removes golint from the `make lint-metrics`target.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/importer/vddk-datasource_test.go [hunk 1]
```diff
@@ -666,14 +666,14 @@ var _ = Describe("VDDK get block status", func() {
 		flagSetting := true
 		currentMockNbdFunctions.BlockStatus = func(length uint64, offset uint64, callback libnbd.ExtentCallback, optargs *libnbd.BlockStatusOptargs) error {
 			err := 0
-			len := uint32(MaxBlockStatusLength)
+			L := uint32(MaxBlockStatusLength)
```
Comment (awels) [inline pkg/importer/vddk-datasource_test.go:669]: why use an upper case L instead of a lower case one?

File: tools/cdi-func-test-proxy/main.go [hunk 2]
```diff
@@ -216,19 +213,10 @@ func main() {
...
-	startServer(httpPortWithAuth, withBasicAuth, false, wg)
-
-	wg.Add(1)
-	startServer(httpsPort, noBasicAuth, true, wg)
-
-	wg.Add(1)
-	startServer(httpsPortWithAuth, withBasicAuth, true, wg)
-
-	wg.Wait()
+	select {}
```
Comment (awels) [inline tools/cdi-func-test-proxy/main.go:221]: This doesn't feel semantically the same as the wait groups. Am I missing something? Also I think the log.Fatal in startServer is probably a mistake on my part.

File: pkg/importer/vddk-datasource_test.go (hunk 1 above)
Comment (EduardGomezEscandell) [reply pkg/importer/vddk-datasource_test.go:669]: The linter dislikes shadowing built-in variables and functions such as `len`, `max`, etc.

The uppercase L is just my lack of originality

File: tools/cdi-func-test-proxy/main.go (hunk 2 above)
Comment (EduardGomezEscandell) [reply tools/cdi-func-test-proxy/main.go:221]: Both things block forever until one of the `startServer` exits the process with `log.Fatal`. The calls to `wg.Done()` are unreachable, so this wg was overkill.

I can rework this to not use log.Fatal and go back to workgroups if you prefer. In that case they wouldn't be overkill.

File: pkg/importer/vddk-datasource_test.go (hunk 1 above)
Comment (awels) [reply pkg/importer/vddk-datasource_test.go:669]: Right it is just that normally upper cass variables indicate 'public' type variables. Which of course is not a thing inside a function, it is just confusing. I would just use lower case l and we can call it a day.

Comment (EduardGomezEscandell) [reply pkg/importer/vddk-datasource_test.go:669]: Sounds good.

**Review by awels — APPROVED** [review]
/approve

**Pull request merged**